elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/core/ViewDictionary.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.datastores.aggregation.queryengines.sql.core;
+
+import com.yahoo.elide.Injector;
+import com.yahoo.elide.core.EntityBinding;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.exceptions.DuplicateMappingException;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.annotation.View;
+import com.yahoo.elide.security.checks.Check;
+
+import com.google.common.collect.Sets;
+import org.apache.commons.lang3.StringUtils;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import javax.persistence.Entity;
+
+/**
+ * Extension of {@link EntityDictionary} that manage view models.
+ */
+@Slf4j
+public class ViewDictionary extends EntityDictionary {
+    private final ConcurrentHashMap<Class<?>, ViewBinding> viewBindings = new ConcurrentHashMap<>();
+    private final static ViewBinding EMPTY_BINDING = new ViewBinding();
+
+    public ViewDictionary(Map<String, Class<? extends Check>> checks) {
```

Comment:
Maybe the only constructor for ViewDictionary should just take another dictionary that it copies.